death-causes-app/src/models/Factors.ts

```diff
@@ -1,70 +1,237 @@
+          //     );
+          //     break;
+          //   }
+          case "string": {
+            this.factorList[element.factorName] = new StringFactorPermanent(
+              element.factorName as string,
+              element.initialValue as string,
+              element.phrasing,
+              element.placeholder,
+              element.options?.split("_")
+            );
+            break;
+          }
+          default:
+            break;
+        }
+      }
+    });
+    console.log(this.factorList);
+  }
 
-    getFactorsAsStateObject() {
-        let stateObject: FactorAnswers = {};
-        this.factorList.forEach(element => { return stateObject[element.factorName] = element.getInitialValue() })
-        return stateObject;
+  getFactorsAsStateObject() {
+    let stateObject: FactorAnswers = {};
+    for (let factorName in this.factorList) {
+      stateObject[factorName] = this.factorList[factorName].getInitialValue();
     }
+    return stateObject;
+  }
+
+  getInputValidity(name: string, value: string | boolean): InputValidity {
+    return this.factorList[name].check_input(value);
+  }
 }
 
-abstract class GeneralFactor<T> {
-    factorName: string;
-    initialValue: T;
+interface ExplanationAndLimits {
+  lowerLim: null | number;
+  upperLim: null | number;
+  explanation: string;
+}
 
-    constructor(factorName: string, initialValue: T) {
-        this.factorName = factorName;
-        this.initialValue = initialValue;
-    }
+enum ExplanationStart {
+  RECOMMENDATION_PHRASING = "More accurate if",
+  REQUIREMENT_PHRASING = "Should be",
+}
 
-    getInitialValue(): T {
-        return this.initialValue;
+function extractLimsAndExplanation(
+  domain: string | null,
+  explanationStart: ExplanationStart
+): ExplanationAndLimits {
+  if (domain === null) {
+    return { lowerLim: null, upperLim: null, explanation: "" };
+  } else {
+    let lowerLim: null | number = null;
+    let upperLim: null | number = null;
+    let explanation: string = "";
+    let a = domain.split("-");
+    if (a[0] !== "") {
+      explanation += explanationStart;
+      lowerLim = parseFloat(a[0]);
+      if (a[1] !== "") {
+        upperLim = parseFloat(a[1]);
+        explanation += " between " + a[0] + " and " + a[1];
+      } else {
+        explanation += " larger than " + a[0];
+      }
+    } else {
+      if (a[1] !== "") {
+        explanation += explanationStart + " smaller than " + a[1];
+        upperLim = parseFloat(a[1]);
+      }
     }
+    return { lowerLim: lowerLim, upperLim: upperLim, explanation: explanation };
+  }
 }
 
-class NumericFactorPermanent extends GeneralFactor<number> {
-    constructor(factorName: string) {
-        super(factorName, 0);
-    }
+abstract class GeneralFactor<T> {
+  factorName: string;
+  initialValue: T | "";
+  phrasing: string; //If the factor is not going to be asked, the phrasing should be nu
+  placeholder: string;
+  factorType: string = "abstract";
```

Comment:
har vi brug "abstract"?

Reply:
factorType brokker sig hvis ikke den bliver initialiseret. Men jeg er begyndt at overveje om det overhovedet kan betale sig at have den abstrakte klasse GeneralFactor<T> - det er træls at en parameter som options, der kun giver mening i StringQuestion, også skal defineres i GeneralFactor.

Reply:
men så lad os fjerne den fra generalfactor og placere den i string question. det er der vel ikke ngoet i vejen for?